LifeCycleAssessment_Engine/Compute/EvaluateEnvironmentalProductDeclaration.cs

```diff
@@ -0,0 +1,194 @@
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+//using BH.oM.Structure; <--This was used for IMass calc. Remove when IelementM is implemented.
+using System.ComponentModel;
+using BH.oM.Base;
+using BH.oM.Reflection.Attributes;
+using BH.oM.LifeCycleAssessment;
+using BH.oM.Quantities.Attributes;
+using BH.Engine.Reflection;
+
+namespace BH.Engine.LifeCycleAssessment
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /****   Public Methods                          ****/
+        /***************************************************/
+
+        [Description("This method calculates the quantity of any selected metric within an Environmental Product Declaration by extracting the declared unit of the selected material and multiplying the objects Volume * Density * EnvironmentalProductDeclarationField criteria. Please view the EnvironmentalProductDeclarationField Enum to explore current evaluation metric options.")]
+        [Input("BHoM Object", "This is a BHoM Object to calculate EPD off of. The method requires a volume property on the BHoM Object. Density is required if the chosen EPD is on a per mass basis, and will be extracted from the dataset if possible prior to extracting from the object itself.")]
+        [Input("environmentalProductDeclaration", "This is LifeCycleAssessment.EnvironmentalProductDeclaration data. Please select your desired dataset and supply your material choice to the corresponding BHoM objects.")]
+        [Input("environmentalProductDeclarationField", "Filter the provided EnvironmentalProductDeclaration by selecting one of the provided metrics for calculation. This method also accepts multiple fields simultaneously.")]
+        [Output("quantity", "The quantity of the desired metric provided by the EnvironmentalProductDeclarationField")]
+        public static double EvaluateEnvironmentalProductDeclarationPerObject(IBHoMObject obj, EnvironmentalProductDeclaration environmentalProductDeclaration, EnvironmentalProductDeclarationField environmentalProductDeclarationField = EnvironmentalProductDeclarationField.GlobalWarmingPotential) //default to globalWarmingPotential evaluation
+        {
+            if (obj != null)
+            {
+                //First check validity of inputs
+                object vol = obj.PropertyValue("Volume");
+                if (vol == null)
+                {
+                    BH.Engine.Reflection.Compute.RecordError("The input object must have a Volume property for this method to work.");
+                    return 0;
+                }
+                double volume = System.Convert.ToDouble(vol);
+                if (volume == 0)
+                {
+                    BH.Engine.Reflection.Compute.RecordError("The input object's Volume value is invalid. Volume should be in m3 in numerical format.");
```

Comment:
Is a negative m3 valid? I would imagine not given volume ought to always be a positive number, so should we change the check to be `if(volume <= 0)` instead and amend the error message accordingly?